File: app/src/main/java/com/temperoni/recipes/mvp/model/RecipesListModel.java
package com.temperoni.recipes.mvp.model;

import com.temperoni.recipes.manager.RecipesManager;
import com.temperoni.recipes.domain.dto.Recipe;
import com.temperoni.recipes.ui.models.RecipeViewModel;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

/**
 * @author Leandro Temperoni
 */
public class RecipesListModel {

    private RecipesManager recipesManager;

    @Inject
    RecipesListModel(RecipesManager recipesManager) {
        this.recipesManager = recipesManager;
    }

    public void fetchData() {
        recipesManager.fetchRecipes();
    }

    public List<RecipeViewModel> getViewModelList(List<Recipe> recipes) {
        List<RecipeViewModel> recipeViewModels = new ArrayList<>();

        for (Recipe recipe : recipes) {
            RecipeViewModel recipeViewModel = new RecipeViewModel(
                    recipe.getId(),
                    recipe.getName(),
                    recipe.getImage());

            recipeViewModels.add(recipeViewModel);
        }

        return recipeViewModels;
    }
}

File: app/src/main/java/com/temperoni/recipes/mvp/model/RecipesListModel.kt
package com.temperoni.recipes.mvp.model

import com.temperoni.recipes.domain.dto.Recipe
import com.temperoni.recipes.manager.RecipesManager
import com.temperoni.recipes.ui.models.RecipeViewModel
import javax.inject.Inject

/**
 * @author Leandro Temperoni
 */
class RecipesListModel
@Inject
constructor(private val recipesManager: RecipesManager) {

    fun fetchData() {
        recipesManager.fetchRecipes()
    }

    fun getViewModelList(recipes: List<Recipe?>?): List<RecipeViewModel> {
        val recipeViewModels = mutableListOf<RecipeViewModel>()

        recipes?.forEach {
            recipeViewModels.add(RecipeViewModel(it?.id ?: 0,
                    it?.name ?: "",
                    it?.image ?: ""))
        }

        return recipeViewModels
    }
}
